### ISSUE TITLE
Nested game fields

### ISSUE THREAD

**Issue by JosefNemec** [JosefNemec/Playnite#2912]
### Feature description

For more granular library organization. For example you would be able to create genre field that has other genre under it:
- FPS
  - Boomer
  - Modern military
- RPG
  - JRPG
  - CRPG

This should then be visualized with additional (nested) groups on game lists, explorer panel and filter panel.

**Comment by mixels** [comment]
This would be awesome. I understand it's a lot of work, but this would provide fantastic possibilities for organizing. It would also enable useful displays of I formation if 0 to infinity nested groups were possible, like:

* Installation Status
  * Install Drive
    * Install Size

**Comment by Fooughhy** [comment]
Would it be possible to add secondary sorting to this as well?

So if you sort by e.g. "Publisher", there could be multiple games by the same publisher. So a secondary sorting parameter would decide how these items should be sorted.

Publisher: Ubisoft
Games:
Assassins Creed Odyssey
Assassins Creed Origins

If you sort by publisher as the main sorting parameter, I am assuming that the secondary sorting parameter is always "Alphabetical" implicitly (even though we can't change it at the moment). It seems to be deterministic at least. This leads to the order:

> Assassins Creed Odyssey
> Assassins Creed Origins

If you introduce secondary sorting parameter, and set it to e.g. "Release Date", the order would be:

> Assassins Creed Origins
> Assassins Creed Odyssey

IMHO it may not be a big difference between adding a single secondary sorting parameter and also being able to add even more.

The end result I think would be best is for us to be able to add multiple sorting parameters and also order them in the way of precedence. So we could easily swap the 1st and 3rd sorting parameter on the fly in our filters, for example.

**Comment by Mertica** [comment]
Nested categories would be perfect to have

**Comment by Bolthier** [comment]
I would really like this feature. I was about to open a similar request, but I think this task should cover it.

In my scenario I like to give categories and tags an additional value. For example I have a category for my Game of the Years, where I can filter just for GOTY and see all of them. What I would prefer would be to have a category with a value or sub-category like "GOTY::2023", "GOTY::2022" and so on. This way I would be able to filter only after "GOTY" and still see all of these games.

**Comment by KrivenRaven** [comment]
![image](https://github.com/JosefNemec/Playnite/assets/43152557/4d6c94c0-5d1b-4660-a50a-8d79a6c90a14)

My body is ready.

**Comment by Bo9da** [comment]
Yes please

**Comment by LemuArtworks** [comment]
This would be really nice to have on series as well, in order to be able to nest something like this way:

- DC Universe
  - Batman
    - Arkham
    - LEGO
    - Telltale
  - Injustice
  - Superman
- Marvel Universe
  - Avengers
    - LEGO
  - Spider-Man
    - Marvel's Spider-Man
    - The Amazing Spider-Man

Multiple nest within nests allow for further customizations and makes things easier to find since you can specify a lot more.
Especially when it comes to huge franchises like DC and Marvel as an example, that has a lot of different series within it's franchise and helps a lot for those with very large libraries.

I haven't really seen any other frontends put this to use, so if you were to allow that amount of organization freedom, it could make playnite stand out a bit more from other frontends such as Launchbox too.

**Comment by mr-falafel** [comment]
I would like to throw my voice in to this pile. Not being able to at least sort by Developer > Release Date is painful. I would prefer to be able to sort by multiple criteria. For example, Platform > Genre > Developer > Release Date > Title.
